etcdmgmt/etcd-mgmt.go

```diff
@@ -160,3 +147,115 @@ func isETCDStartNeeded() (bool, int) {
 	}
 	return start, pid
 }
+
+func EtcdStartInit() (*os.Process, error) {
+	// Check whether etcd environment variable present or not
+	// If it present then start etcd without --initial-cluster flag
+	// other wise start etcd normally.
+
+	file, err := os.Open(etcdConfFile)
+	if err != nil {
+		log.Info("Starting/Restarting etcd for a initial node")
+		return StartInitialEtcd()
+	} else {
+		defer file.Close()
+
+		// Restoring etcd environment variable and starting etcd daemon
+		scanner := bufio.NewScanner(file)
+		scanner.Split(bufio.ScanLines)
+		for scanner.Scan() {
+			linestr := scanner.Text()
+			etcdenv := strings.Split(linestr, "=")
+
+			etcdEnvKey := etcdenv[0]
+			etcdEnvData := etcdenv[1]
+			envlen := len(etcdenv)
+			for i := 2; i < envlen; i++ {
+				etcdEnvData = etcdEnvData + "=" + etcdenv[i]
+			}
+
+			// setting etcd environment variable
+			os.Setenv(etcdEnvKey, etcdEnvData)
+		}
+
+		hostname, err := os.Hostname()
+		if err != nil {
+			log.Fatal("Could not able to get hostname")
+			return nil, err
+		}
+
+		listenClientUrls := "http://" + hostname + ":2379"
+		advClientUrls := "http://" + hostname + ":2379"
+		listenPeerUrls := "http://" + hostname + ":2380"
+		initialAdvPeerUrls := "http://" + hostname + ":2380"
+
+		args := []string{"-listen-client-urls", listenClientUrls,
+			"-advertise-client-urls", advClientUrls,
+			"-listen-peer-urls", listenPeerUrls,
+			"-initial-advertise-peer-urls", initialAdvPeerUrls}
```

Comment:
I can see this snippet of forming args is been used in multiple places, so definitely can be reusable with the help of a function.